### DISCUSSION TITLE
Current implementation of PG Range.__bool__ is surprising

### DISCUSSION THREAD

**Discussion by lelit** [sqlalchemy/sqlalchemy#8883] [Ideas]
What is the rationale behind [current implementation](https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/dialects/postgresql/ranges.py#L84) of the `__bool__` method of the new PG Range class?

It returns `True` when the range is *empty*, `False` otherwise: IMHO, in Python, one would expect the other way around.

In the principle of least surprise, I'd reverse its meaning.

What do you think?

**Comment by lelit** [comment]
FWIW, both [asyncpg](https://github.com/MagicStack/asyncpg/blob/master/asyncpg/types.py#L121) and [psycopg](https://github.com/psycopg/psycopg/blob/master/psycopg/psycopg/types/range.py#L163) implement that method with the inverse logic suggested above.

**Comment by zzzeek** [comment]
it's a typo.  I was almost certainly copying [this](https://github.com/MagicStack/asyncpg/blob/43bd82c54e29352ae08b1982c23b9cb0cc33717d/asyncpg/types.py#L121)

**Reply by lelit** [reply]
Great, thanks. Does this warrant an issue and a PR?

**Reply by zzzeek** [reply]
if you have the time, it would be helpful if you could send a PR.  no issue required as we haven't released the most recent set of range changes in any case.
